flume-ng-sinks/flume-hdfs-sink/src/main/java/org/apache/flume/sink/hdfs/HDFSEventSink.java

```diff
@@ -354,9 +354,9 @@ static CompressionCodec getCodec(String codecName) {
   public Status process() throws EventDeliveryException {
     Channel channel = getChannel();
     Transaction transaction = channel.getTransaction();
-    List<BucketWriter> writers = Lists.newArrayList();
     transaction.begin();
     try {
+      Set<BucketWriter> writers = new LinkedHashSet<>();
```

Comment:
I'm not sure but I vaguely remember some issue regarding to the ordering (cc. @simonati) so I would keep the `LinkedHashSet`. On the other hand not tests break with `HashSet`.